rest/indextest/index_test.go

```diff
@@ -596,6 +596,226 @@ func TestSyncOnline(t *testing.T) {
 
 }
 
+// TestAsyncInitConfigUpdates verifies that a database with in-progress async
+// index initialization can accept all config updates from the local node.
+// (prior to CBG-4008, operations would block waiting for async init to complete)
+func TestAsyncInitConfigUpdates(t *testing.T) {
+	if base.UnitTestUrlIsWalrus() {
+		t.Skip("This test only works against Couchbase Server")
+	}
+	base.TestRequiresCollections(t)
+	base.SetUpTestLogging(t, base.LevelDebug, base.KeyHTTP)
+
+	sc, closeFn := rest.StartBootstrapServer(t)
+	defer closeFn()
+
+	// Set testing callbacks for async initialization
```

Comment:
```suggestion
	// Set testing callbacks for async initialization to pause after a single collection's indexes have been created.
```

We could actually shorten this test timeout to only create a single named collection and speed up these tests, since it will loop once for named collection and the default collection. This can make a pretty significant speedup over a bunch of tests, see https://github.com/couchbase/sync_gateway/pull/6911

Reply:
Do you think this merits this being handled differently than the rest of our test suite (i.e. using SG_TEST_COLLECTION_POOL_SIZE)?